packages/core/elasticsearch/core-elasticsearch-client-server-internal/src/agent_manager.ts

```diff
@@ -69,8 +78,16 @@ export class AgentManager implements AgentFactoryProvider, AgentStore {
     };
   }
 
-  public getAgents(): Set<NetworkAgent> {
-    return this.agents;
+  public getAgentsStats(): ElasticsearchClientsMetrics {
+    const stats = getAgentsSocketsStats(this.agents);
+
+    if (stats.totalQueuedRequests > 0) {
+      this.logger.warn(
+        `There are ${stats.totalQueuedRequests} queued requests. If this number is constantly high, consider scaling Kibana horizontally or increasing "elasticsearch.maxSockets" in the config.`
+      );
+    }
+
+    return stats;
```

Comment:
I moved the calculation logic closer to the Agent Manager so that we didn't need to expose the logger publicly.

At the same time, it feels safer if we are not exposing the agents publicly for anyone else to tamper with.